Makefile

```diff
@@ -31,15 +31,15 @@ TARGETS := \
 
 # Build all ELF binaries using a Dockerized LLVM toolchain.
 docker-all:
-	docker run --rm --user "${UIDGID}" \
+	docker run --platform linux/amd64 --rm --user "${UIDGID}" \
```

Comment:
Unfortunately this breaks on my machine:
```
$ make
docker run --platform linux/amd64 --rm --user "1000:1000" \
	-v "/home/lorenz/dev/ebpf":/ebpf -w /ebpf --env MAKEFLAGS \
	--env CFLAGS="-fdebug-prefix-map=/ebpf=." \
	"quay.io/cilium/ebpf-builder:1619686610" \
	make all
"--platform" is only supported on a Docker daemon with experimental features enabled
make: *** [Makefile:34: docker-all] Error 1
```

Reply:
This change feels a little unnecessary. `docker run` should try to pull and run an image compatible with the host's architecture. (I assume this is the point of distributing arm64 LLVM builder images) Does this actually work on M1? Will it try to pull and run an x86 Docker image and execute it on an emulated x86 host?

If so, what's the point of building arm64 toolchain images? (I assume it's horribly slow for larger eBPF codebases)

Reply:
I had a quick look the other day, I'm not sure docker is smart enough unfortunately. Something about us having to publish a manifest. I didn't dig too deep and would love to proven wrong.

Reply:
> This change feels a little unnecessary. `docker run` should try to pull and run an image compatible with the host's architecture. (I assume this is the point of distributing arm64 LLVM builder images) Does this actually work on M1? Will it try to pull and run an x86 Docker image and execute it on an emulated x86 host?
> 
> If so, what's the point of building arm64 toolchain images? (I assume it's horribly slow for larger eBPF codebases)

I have tried this one on my M1 in both Docker Desktop for Mac and my Parallels Ubuntu 20.04 VM. Both work with this flag, and I would say is usable for the needs of the development within this project. At this point, I don't think there is any reason to build arm64 images at all.